river/InputConfig.zig

```diff
@@ -232,9 +261,13 @@ tap: ?TapState = null,
 @"pointer-accel": ?PointerAccel = null,
 @"scroll-method": ?ScrollMethod = null,
 @"scroll-button": ?ScrollButton = null,
+@"map-to-output": ?MapToOutput = null,
```

Comment:
What's the difference between this field being null and `MapToOutput.output_name` being null? it seems to me that there is no difference and they shouldn't both be optional.

Note that for the libinput config options we use a null field here to indicate that the default value from libinput is used which may depend on hardware/libinput version/whatever. That's not the case here though, the default is no mapping and is determined by us.

Reply:
The reason I decided to do it this way is that, with your suggestion, the `apply` function wouldn't be called in case the output is null. This would result in outputs no longer being explicitly unmapped, breaking a `map-output disabled` on devices that are currently mapped as well as unmapping on input disconnect.

Reply:
I don't follow.

Whether the apply function is called solely depends on whether the field is set, not on what outputs are present either way, no?

Reply:
Yes, but suppose a user does `map-to-output DP-1`. The field is now set to a non-null value, and when apply is called afterwards, the input device is correctly mapped. Afterwards, the user may run `map-to-output disabled`. The field is now set to null, and nothing happens once the `InputConfig` is applied, preventing the device from being unmapped again.

What I meant when I said "the output is null" earlier is that the `output_name` field is null.